pkg/fanal/analyzer/language/nodejs/yarn/yarn.go

```diff
@@ -78,8 +101,40 @@ func (a yarnAnalyzer) PostAnalyze(_ context.Context, input analyzer.PostAnalysis
 }
 
 func (a yarnAnalyzer) Required(filePath string, _ os.FileInfo) bool {
-	fileName := filepath.Base(filePath)
-	return fileName == types.YarnLock || fileName == types.NpmPkg
+	dirs, fileName := splitPath(filePath)
+
+	if fileName == types.YarnLock &&
+		containsAny(filePath, "node_modules", ".yarn/unplugged") {
+		return false
+	}
+
+	if fileName == types.YarnLock ||
+		fileName == types.NpmPkg ||
+		strings.HasPrefix(strings.ToLower(fileName), "license") {
```

Comment:
if I understand correctly - any license file name is possible:
https://github.com/aquasecurity/trivy/blob/ba38c06c32db1c091d9bb26fa43a742cecbd94c1/pkg/fanal/analyzer/language/nodejs/license.go#L79-L86
Perhaps we need to save `node_modules`/`.yarn` folder?

Reply:
The cache folder can be large.

Reply:
I assumed that...
Okay. I think in most cases license file name contains `license`. 
Let's wait for feedback from users.
If users say that there are files with other names - we will return to this question.